modules/sdk/gradle-plugins-node/src/main/java/com/liferay/gradle/plugins/node/tasks/NpmRunTask.java

```diff
@@ -0,0 +1,286 @@
+
+		File file = project.file("package-lock.json");
+
+		if (!file.exists()) {
+			return null;
+		}
+
+		return file;
+	}
+
+	@Input
+	public String getScriptName() {
+		return GradleUtil.toString(_scriptName);
+	}
+
+	@InputFile
+	@Optional
+	public File getShrinkwrapJsonFile() {
+		Project project = getProject();
+
+		File file = project.file("npm-shrinkwrap.json");
+
+		if (!file.exists()) {
+			return null;
+		}
+
+		return file;
+	}
+
+	@OutputFile
+	public File getSourceDigestFile() {
+		Project project = getProject();
+
+		String pathname = "npm/script/" + getName() + "/.digest";
+
+		return new File(project.getBuildDir(), pathname);
+	}
+
+	@Input
+	@Optional
+	public File getSourceDir() {
+		return GradleUtil.toFile(getProject(), _sourceDir);
+	}
+
+	@InputFiles
+	@Optional
+	public FileCollection getSourceFiles() {
+		File sourceDir = getSourceDir();
+
+		if ((sourceDir == null) || !sourceDir.exists()) {
+			return null;
+		}
+
+		Project project = getProject();
+
+		FileTree fileTree = project.fileTree(sourceDir);
+
+		FileCollection fileCollection = fileTree.matching(_patternFilterable);
+
+		if (fileCollection.isEmpty()) {
+			return null;
+		}
+
+		return fileCollection;
+	}
+
+	@Override
+	public NpmRunTask include(
+		@SuppressWarnings("rawtypes") Closure includeSpec) {
+
+		_patternFilterable.include(includeSpec);
+
+		return this;
+	}
+
+	@Override
+	public NpmRunTask include(Iterable<String> includes) {
+		_patternFilterable.include(includes);
+
+		return this;
+	}
+
+	@Override
+	public NpmRunTask include(Spec<FileTreeElement> includeSpec) {
+		_patternFilterable.include(includeSpec);
+
+		return this;
+	}
+
+	@Override
+	public NpmRunTask include(String... includes) {
+		_patternFilterable.include(includes);
+
+		return this;
+	}
+
+	@Override
+	public NpmRunTask setExcludes(Iterable<String> excludes) {
+		_patternFilterable.setExcludes(excludes);
+
+		return this;
+	}
+
+	@Override
+	public NpmRunTask setIncludes(Iterable<String> includes) {
+		_patternFilterable.setIncludes(includes);
+
+		return this;
+	}
+
+	public void setNodeVersion(Object nodeVersion) {
+		_nodeVersion = nodeVersion;
+	}
+
+	public void setNpmVersion(Object npmVersion) {
+		_npmVersion = npmVersion;
+	}
+
+	public void setScriptName(Object scriptName) {
+		_scriptName = scriptName;
+	}
+
+	public void setSourceDir(Object sourceDir) {
+		_sourceDir = sourceDir;
+	}
+
+	@Override
+	protected List<String> getCompleteArgs() {
+		List<String> completeArgs = super.getCompleteArgs();
+
+		completeArgs.add("run-script");
+		completeArgs.add(getScriptName());
+
+		return completeArgs;
+	}
+
+	private void _writeSourceDigestFile(byte[] bytes) throws Exception {
+		File file = getSourceDigestFile();
+
+		Files.write(file.toPath(), bytes);
+	}
+
+	private static final String[] _EXCLUDE_DIR_NAMES = {
+		"bin", "build", "classes", "node_modules", "test-classes", "tmp"
+	};
+
+	private static final String[] _INCLUDES = {
+		"**/*.css", "**/*.js", "**/*.*rc"
```

Comment:
You'll probably want to also add a check for `**/*.json` and `**/*.jsx`. I forgot we use the .jsx extension and also since .json can be imported to js files, we should also check that as well.